### DISCUSSION TITLE
Nextra with yaml support?

### DISCUSSION THREAD

**Discussion by prensoni0143** [shuding/nextra#2697] [Q&A]
Hi Team, does Nextra supports YAML file for page rendering?

**Comment by caffeine-addictt** [comment]
Can you provide more context?

Not too sure about generating pages at runtime, but if you can try adding a build step that generates the pages from yaml following Nextra's file structure.

**Reply by prensoni0143** [reply]
xxx edit

**Reply by caffeine-addictt** [reply]
Looks like you'll have to build and `npm start` every time you change something locally. 🥲

**Reply by prensoni0143** [reply]
Sorry, please ignore my above comment, it was meant for another issue I reported.

To describe the ask above, we have yaml files from which we need to generate static page content. I found the way by using yaml-parsing with API hosting to convert it to json, however I would prefer directly to be able to do it. below is the code of what I am trying:

`import axios from "axios";
import { useEffect, useState } from "react";


export default function Controls({ file, initialControls }) {
  const [controls, setControls] = useState(initialControls || []);

useEffect(() => {
if (!initialControls) {
fetchData();
}
}, []);

async function fetchData() {
try {
const response = await axios.get(
`https://freecodez-api.onrender.com/api/yaml-parsing?file=./data/${file}`
);
if (response?.data?.controls) {
setControls(response.data.controls);
}
} catch (error) {
console.error("Error fetching data:", error);
}
}

return (
<div>
{controls.map((control) => (
<div key={control.name}>
<h3>
  <a href={`#${control.name}`}>{control.description}</a>
</h3>
<p>Control Name: {control.name}</p>
<p>Control Severity: {control.score}</p>
<p>
Control Query:{" "}
<code>
{control.query.replace("&#39;", "'").replace("&quot;", '"')}
</code>
</p>
<p>Enabled status: {control.enabled}</p>
<p>Control Category: {control.category}</p>
<p>`

**Reply by prensoni0143** [reply]
and the file where it should generate has the follow:

`import Controls from "../../../components/Controls.mdx";

<Controls file={"alicloud-controls.yaml"} />`

**Reply by prensoni0143** [reply]
My ask here if I can use directly yaml file to parsh the content without converting it to json, it would be really helpful.

**Reply by caffeine-addictt** [reply]
I don't think rendering directly from yaml is supported in Nextra/NextJS. This is more so of a hack, but something like this could work. [NextJS Docs](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props)

```tsx
import * as React from 'react'
import type { GetServerSidePropsContext } from 'next'

interface ExportedContent { content: any; orEvenAsAPage: React.ReactElement | react.ReactElement[] }

export const getServerSideProps = ({ res }: GetServerSidePropsContext): Promise<{ props: ExportedContent }> => {
    // Code here runs on the server
    // You can import the file and interpret it to JSX/Javascript

    return { props: { content: interpretedYamlStuff, orEvenAsAPage: page } }
}


const MyPage = ({ props: { content, Page } }: { props: ExportedContent }) => {
    // Client code here
    return <>
        {content}
        <Page />
    </>
}

```
